test/python/algorithms/optimizers/test_optimizers.py

```diff
@@ -155,5 +171,215 @@ def callback(x):
         self.assertTrue(values)  # Check the list is nonempty.
 
 
+@ddt
+class TestOptimizerSerialization(QiskitAlgorithmsTestCase):
+    """Tests concerning the serialization of optimizers."""
+
+    @data(
+        ("BFGS", {"maxiter": 100, "eps": np.array([0.1])}),
+        ("CG", {"maxiter": 100, "gtol": 1e-8}),
+        ("COBYLA", {"maxiter": 100}),
+        ("L_BFGS_B", {"maxiter": 100}),
+        ("NELDER_MEAD", {"maxiter": 100}),
+        ("NFT", {"maxiter": 100}),
+        ("P_BFGS", {"maxiter": 100}),
+        ("POWELL", {"maxiter": 100}),
+        ("SLSQP", {"maxiter": 100}),
+        ("TNC", {"maxiter": 100}),
+        ("dogleg", {"maxiter": 100}),
+        ("trust-constr", {"maxiter": 100}),
+        ("trust-ncg", {"maxiter": 100}),
+        ("trust-exact", {"maxiter": 100}),
+        ("trust-krylov", {"maxiter": 100}),
+    )
+    @unpack
+    def test_scipy(self, method, options):
+        """Test the SciPyOptimizer is serializable."""
+
+        optimizer = SciPyOptimizer(method, options=options)
+        serialized = optimizer.to_dict()
+        from_dict = SciPyOptimizer.from_dict(serialized)
+
+        self.assertEqual(from_dict._method, method.lower())
+        self.assertEqual(from_dict._options, options)
+
+    def test_scipy_not_serializable(self):
+        """Test serialization fails if the optimizer contains an attribute that's not supported."""
+
+        def callback(x):
+            print(x)
+
+        optimizer = SciPyOptimizer("BFGS", options={"maxiter": 1}, callback=callback)
+
+        with self.assertRaises(QiskitError):
+            _ = optimizer.to_dict()
+
+    def test_scipy_name_missing(self):
```

Comment:
How about invalid name test if my name is incorrect or does not match an installed optimizer?